fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergTransaction.java

```diff
@@ -59,140 +66,114 @@ public void updateIcebergCommitData(List<TIcebergCommitData> commitDataList) {
-        AppendFiles appendFiles = transaction.newAppend();
-
-        for (CommitTaskData task : convertToCommitTaskData()) {
-            DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
-                    .withPath(task.getPath())
-                    .withFileSizeInBytes(task.getFileSizeInBytes())
-                    .withFormat(IcebergUtils.getFileFormat(icebergTable))
-                    .withMetrics(task.getMetrics());
-
-            if (icebergTable.spec().isPartitioned()) {
-                List<String> partitionValues = task.getPartitionValues()
-                        .orElseThrow(() -> new VerifyException("No partition data for partitioned table"));
-                builder.withPartitionValues(partitionValues);
-            }
-            appendFiles.appendFile(builder.build());
-        }
+    public void preCommit(SimpleTableInfo tableInfo, Optional<InsertCommandContext> insertCtx) {
 
-        // in appendFiles.commit, it will generate metadata(manifest and snapshot)
-        // after appendFiles.commit, in current transaction, you can already see the new snapshot
-        appendFiles.commit();
-    }
+        LOG.info("iceberg table {} insert table finished!", tableInfo);
 
-    public List<CommitTaskData> convertToCommitTaskData() {
-        List<CommitTaskData> commitTaskData = new ArrayList<>();
-        for (TIcebergCommitData data : this.commitDataList) {
-            commitTaskData.add(new CommitTaskData(
-                    data.getFilePath(),
-                    data.getFileSize(),
-                    new Metrics(
-                            data.getRowCount(),
-                            Collections.EMPTY_MAP,
-                            Collections.EMPTY_MAP,
-                            Collections.EMPTY_MAP,
-                            Collections.EMPTY_MAP
-                    ),
-                    data.isSetPartitionValues() ? Optional.of(data.getPartitionValues()) : Optional.empty(),
-                    convertToFileContent(data.getFileContent()),
-                    data.isSetReferencedDataFiles() ? Optional.of(data.getReferencedDataFiles()) : Optional.empty()
-            ));
+        //create  start the  iceberg transaction
+        TUpdateMode updateMode = TUpdateMode.APPEND;
+        if (insertCtx.isPresent()) {
+            updateMode = ((BaseExternalTableInsertCommandContext) insertCtx.get()).isOverwrite() ? TUpdateMode.OVERWRITE
+                    : TUpdateMode.APPEND;
         }
-        return commitTaskData;
+        updateManifestAfterInsert(updateMode);
     }
 
-    private FileContent convertToFileContent(TFileContent content) {
-        if (content.equals(TFileContent.DATA)) {
-            return FileContent.DATA;
-        } else if (content.equals(TFileContent.POSITION_DELETES)) {
-            return FileContent.POSITION_DELETES;
+    private void updateManifestAfterInsert(TUpdateMode updateMode) {
+
+        Table table = getNativeTable(tableInfo);
```

Comment:
Transactions are isolated and cache tables cannot be used

Reply:
First, the getNativeTable function gets the table from IcebergUtil. IcebergUtil is a tool class that is not only used in the current scenario.
Secondly, we should also need to get the latest information of the table.